### PR TITLE
Blog page including Potion API and Tilt.js library

### PR DESCRIPTION
**Pull request by cyu6**
Potion API: https://benborgers.com/blog/notion-to-website
Tilt.js: https://gijsroge.github.io/tilt.js/
Screenshot:
blog

### CODE REVIEW

**Review by hehhjiang — APPROVED** [review]

File: portfolio/src/main/webapp/blog.html [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        </div> 
+        <div class="post" data-tilt>
+          <h3>title</h3>
+          <h4>date written</h4>
+          <p>some text about the post...</p>         
+        </div> 
+      </div> 
+    </div>
+
+    <script src="https://code.jquery.com/jquery-3.5.1.min.js"></script>
```
Comment (paulaksp) [inline portfolio/src/main/webapp/blog.html:48]: nit: ccould we have script srcs go all in the same place above, and you can comment it with section  or something similar.

File: portfolio/src/main/webapp/projects.html [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      <div class="project-block" id="cdpush">
+        <h3>Config Cleanup @ Google</h3>
+        <h4>June - September 2019 | Engineering Practicum Intern</h4> 
+        <p>CDPush team in Sunnyvale. Developed a C++ server to scan and clean up files in a storage system.</p>
+        <p>Used C++, learned about Go, distributed systems, file management systems</p> 
+      </div>
+      <h2>I've recently become interested in UI/UX design:</h2>
+      <ul>
+        <li>MIT Beat the Pandemic II Project: <a href="">link to wireframe</a></li> 
+        <li>KoKo Pops Design Team Member: <a href="">link to work</a></li>
```
Comment (paulaksp) [inline portfolio/src/main/webapp/projects.html:39]: nit: i think for things that are not yet completed, even during development might be good practice to start adding TODOs. For the HTML, you can just add the , etc.

Just looking ahead--not sure how this would work later with the capstone / github since TODOs can't be linked up with the buganizer, but there should be some equivalent that we can set-up later with issue #'s.

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -48,3 +48,9 @@ function addRandomCat() {
   if (prev !== undefined)  catContainer.replaceChild(catimg, prev);
   else catContainer.appendChild(catimg);
 }
+
+const postId1 = "0cb628857f3c4c77bf7f9a879a6ec21d";
+fetch("https://potion-api.now.sh/html?id=" + postId1)
+    .then(res => res.text())
+    .then(text => { document.getElementById("post1").innerHTML = text; 
+});
```
Comment (paulaksp) [inline portfolio/src/main/webapp/script.js]: (1) does this have to be in the global scope? Could we have this inside a window.onload function, and that way the 'fetch' call happens once the window is loaded in javascript. 

(2) I realized with github i couldnt comment on unchanged lines, but just to make it more consistent javascript programming for now and the future, let's start removing the 'function' keywords in the javascript, and use arrow functions instead. This will make it more consistent, and more inline with the general javascript/typescript conventions most people use for best practice.
see style guide if you want to read more about this https://google.github.io/styleguide/jsguide.html#features-functions-arrow-functions.

(3) Structure the javascript files so all the constants are at the top

ie

```
// Imports from other files/libraries/ etc if you're using them, and depending on what 'framework' you might/might not use goes at the top.
// const lib1 = import/requirefromlib1

// Consts in one place
const CAT_IMAGES = []
const GREETING_CHOICES = []
const POST_ID = []

// Add the functions below, with the comment for each of the fn's.
window.onload = () => {
...
}
```

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -32,7 +32,7 @@ function addRandomGreeting() {
  */
 function addRandomCat() {
   const cats =
-      ["/images/kitten-in-bed.jpg", "/images/sleepy-kitten.jpg"];
+      ["images/kitten-in-bed.jpg", "images/sleepy-kitten.jpg"];
```
Comment (paulaksp) [inline portfolio/src/main/webapp/script.js]: for actual constants, with the comment below also capitalize them const CATS vs const cats. This will make it easier to differentiate between the `const greetings` which is the actual 'global' constant, vs `const greeting` which is the calculated greeting in the function.

File: portfolio/src/main/webapp/style.css [hunk 5]
```diff
@@ -9,29 +9,67 @@
   float: left;
 }
 #navbar li a {
+  color: #000000;
```
Comment (paulaksp) [inline portfolio/src/main/webapp/style.css]: nit: #000 prefer to #000000

File: portfolio/src/main/webapp/style.css [hunk 6]
```diff
@@ -9,29 +9,67 @@
...
-  color: black;
-  text-align: center;
   padding: 14px 16px;
+  text-align: center;
   text-decoration: none;
 }
 #navbar li a:hover {
-  background-color: lightcoral;
+  background-color: #006c84;
+  color: #ffffff;
```
Comment (paulaksp) [inline portfolio/src/main/webapp/style.css]: nit: same here with #fff

File: portfolio/src/main/webapp/style.css [hunk 7]
```diff
@@ -9,29 +9,67 @@
...
+  text-decoration: none;
+}
+#blog-content a:visited {
+  color: #000000;
+}
 
+#post1 img {
+  max-width: 100%;
+  max-height: 70vh;
+}
```
Comment (paulaksp) [inline portfolio/src/main/webapp/style.css:140]: With the growing css file since most of them seem to be divided by IDs, you could consider splitting the css files into the page/project-specific css (one css file for each of the pages), with a single common css that runs across all your separate pages as well.

This will make it easier to have the independent PRs, as well as make the css cleaner and more structured, imo.

**Review by paulaksp — CHANGES_REQUESTED** [review]

File: portfolio/src/main/webapp/projects.html (hunk 2 above)
Comment (cyu6) [reply portfolio/src/main/webapp/projects.html:39]: edited this in the projects PR
will keep this in mind for the future!

File: portfolio/src/main/webapp/style.css (hunk 7 above)
Comment (cyu6) [reply portfolio/src/main/webapp/style.css:140]: Made a css file for just the posts folder!

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (cyu6) [reply portfolio/src/main/webapp/script.js]: I think I covered all of these!

File: portfolio/src/main/webapp/blog.html (hunk 1 above)
Comment (cyu6) [reply portfolio/src/main/webapp/blog.html:48]: I think for this library, the script lines have to go at the end. I tried moving them to the top and the effect disappeared, based on their usage docs too it seems like they should be at the end

Comment (paulaksp) [reply portfolio/src/main/webapp/blog.html:48]: hm i see ok.

File: portfolio/src/main/webapp/style.css [hunk 8]
```diff
@@ -56,20 +73,68 @@
...
+#photos-content .photo-block:hover .photo-text {
+  display: block;
 }
+@media screen and (max-width: 800px) {
+  #photos-content .photo-block {
+    flex: 90%;
+    max-width: 90%;
+  }
+}
```
Comment (paulaksp) [inline portfolio/src/main/webapp/style.css]: newline to remove the error icon.

File: portfolio/src/main/webapp/script.js [hunk 9]
```diff
@@ -49,8 +48,9 @@ function addRandomCat() {
...
 
-const postId1 = "0cb628857f3c4c77bf7f9a879a6ec21d";
-fetch("https://potion-api.now.sh/html?id=" + postId1)
+window.onload = () => {
+  fetch("https://potion-api.now.sh/html?id=" + POST_ID)
     .then(res => res.text())
     .then(text => { document.getElementById("post1").innerHTML = text; 
-});
+  });
+}
```
Comment (paulaksp) [inline portfolio/src/main/webapp/script.js]: newline error icon.

File: portfolio/src/main/webapp/posts/posts-style.css [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+  max-width: 100%;
+  max-height: 70vh;
+}
+.post-content {
+  margin-left: auto;
+  margin-right: auto;
+  margin-top: 5%;
+  margin-bottom: 10%;
+  width: 60%;
+}
```
Comment (paulaksp) [inline portfolio/src/main/webapp/posts/posts-style.css]: here too.

**Review by paulaksp — APPROVED** [review]
